lib/inspec/rule.rb

```diff
@@ -267,6 +275,45 @@ def __add_check(describe_or_expect, values, block)
       @__checks.push([describe_or_expect, values, block])
     end
 
+    # Look for an input with a matching ID, and if found, apply waiver
+    # skipping logic. Basically, if we have a current waiver, and it says
+    # to skip, we'll replace all the checks with a dummy check (same as
+    # only_if mechanism)
+    # Double underscore: not intended to be called as part of the DSL
+    def __apply_waivers
+      input_name = "waiver_#{@__rule_id}" # TODO: control ID slugging
+      registry = Inspec::InputRegistry.instance
+      input = registry.inputs_by_profile[@__profile_id].dig(input_name)
+      return unless input
+
+      input = input.value
+
+      # Waivers should have a hash value with keys including skip and
+      # expiration_date. We only care here if it has a skip key and it
+      # is yes-like, since all non-skipped waiver operations are handled
+      # during reporting phase.
+      return unless input.key?("skip")
+      return unless input["skip"].to_s.match(/y|yes|true/i)
```

Comment:
I've been assuming that the job of phase 2 is to handle skipping - alterations to the runtime execution plan - and phase three is to handle reporting.  Both phase 2 and 3 can look to the InputRegistry as the sole source of truth for waiver data. 

I think if we were to stash that data somewhere now, I'd feel like we were just making another copy of it for unclear reasons. Unless we want to avoid scanning the entire list of controls during reporting?

Reply:
Having slept on it and done some more code reading, I think doing the logic once here in step 2, then attaching the data somewhere (the control metadata) seems the most sensible. I've done so on d118ed62.

Reply:
The above is still trying to do the job that YAML already does:

```
9982 % ruby -ryaml -e 'p YAML.load("---\nkey: yes")'
{"key"=>true}
```